### PR TITLE
Add timeout to CRI endpoint connection

### PR DESCRIPTION
**Pull request by jtumber-ibm**

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
This looks reasonable to me and hopefully fixes the issues we've see with hanging on ibm cloud. Thanks @jtumber-ibm!

File: pkg/adaptor/proxy/proxy.go [hunk 1]
```diff
@@ -105,26 +105,27 @@ func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error)
...
+		timeout, cancel := context.WithTimeout(ctx, p.retryInterval)
+		defer cancel()
+		conn, err := grpc.DialContext(timeout, p.criSocketPath, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(),
 			grpc.WithContextDialer(func(ctx context.Context, target string) (net.Conn, error) {
 				return (&net.Dialer{}).DialContext(ctx, "unix", target)
 			}),
 		)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to established cri uds connection to %s", p.criSocketPath)
```
Comment (wainersm) [inline pkg/adaptor/proxy/proxy.go:118]: Could you also print `retryInterval`?

Comment (wainersm) [reply pkg/adaptor/proxy/proxy.go:118]: hmm... ok, it depends on the error type, it might not fail due to the timeout.

**Comment by wainersm** [comment]
@jtumber-ibm `retryInterval` has been used in `AgentProxy.dial()` and now `initCriClient()`. Wouldn't have sense to have a timeout specific to cri client connection? I suppose it would be a smaller timeout because the connection is local, or I got it wrong?

File: pkg/adaptor/proxy/proxy.go (hunk 1 above)
Comment (jtumber-ibm) [reply pkg/adaptor/proxy/proxy.go:118]: This doesn't have any retry logic, as the socket should either be there or not at this point. Unlike the agent-protocol-forwarder connection.

**Comment by jtumber-ibm** [comment]
> @jtumber-ibm `retryInterval` has been used in `AgentProxy.dial()` and now `initCriClient()`. Wouldn't have sense to have a timeout specific to cri client connection? I suppose it would be a smaller timeout because the connection is local, or I got it wrong?

We could reduce the timeout for the cri cilent connection, but I don't see the benefits to be honest. The next step after here is the agent-protocol-forwarder connection which normally takes 20+ seconds for the podvm to come online so the overall performance doesn't get degraded. But I'm not opposed to adding in a different timeout just for cri client.

**Review by wainersm — APPROVED** [review]
LGTM

**Pull request merged**